mempool/src/interface/mempool_interface.rs

```diff
@@ -23,17 +23,23 @@ use subsystem::{CallRequest, ShutdownRequest};
 
 pub trait MempoolInterface: Send + Sync {
     /// Add a transaction to mempool
-    fn add_transaction(&mut self, tx: SignedTransaction) -> Result<(), Error>;
+    fn add_transaction(&mut self, tx: SignedTransaction) -> Result<TxStatus, Error>;
 
     /// Get all transactions from mempool
     fn get_all(&self) -> Result<Vec<SignedTransaction>, Error>;
 
-    /// Get a specific transaction from the mempool
+    /// Get a specific transaction from the main mempool
     fn transaction(&self, id: &Id<Transaction>) -> Result<Option<SignedTransaction>, Error>;
 
-    /// Check given transaction is contained in the mempool
+    /// Get a specific transaction from the orphan pool
+    fn orphan_transaction(&self, id: &Id<Transaction>) -> Result<Option<SignedTransaction>, Error>;
+
+    /// Check given transaction is contained in the main mempool
     fn contains_transaction(&self, tx: &Id<Transaction>) -> Result<bool, Error>;
 
+    /// Check given transaction is contained in the main mempool
+    fn contains_orphan_transaction(&self, tx: &Id<Transaction>) -> Result<bool, Error>;
```

Comment:
I'm wondering whether it's of value to return a value vs return whether it exists...

Reply:
It's being used in a number of places where the value is not required. Functions returning the value are available too.